### ISSUE TITLE
SRT protocol not working in Media Source under Linux

### ISSUE THREAD

**Issue by enyone** [obsproject/obs-studio#5238]
### Operating System Info

Ubuntu 20.04

### OBS Studio Version

27.0.1

### OBS Studio Log URL

https://obsproject.com/logs/jtz6k_dKgBhFblrw

### Expected Behavior

Playing stream using Media Source with srt:// prefix url should work (like as using udp:// prefix url).

### Current Behavior

Playing stream using Media Source with srt:// prefix url does not work.

### Steps to Reproduce

1. Test your URL (with srt:// prefix) with ffplay

If works with ffplay (playing stream works), then..

2. Start OBS
3. Create Media Source in Sources
4. Use URL (with srt:// prefix)
5. Save Media Source

Does not work (playing stream does not work).

**Comment by tt2468** [comment]
I believe FFMpeg on Ubuntu 20.04 is not built with SRT enabled. You may need to build and install FFMpeg yourself with `--enable-libsrt`.

**Comment by enyone** [comment]
I don't have "the stock" ffmpeg from Ubuntu repositories installed (only a self-compiled ffmpeg is installed).

This is the output of `ffmpeg -protocols` from the machine.
```
ffmpeg version 4.4 Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 9 (Ubuntu 9.3.0-17ubuntu1~20.04)
  configuration: --prefix=/root/ffmpeg_build --pkg-config-flags=--static --extra-cflags=-I/root/ffmpeg_build/include --extra-ldflags=-L/root/ffmpeg_build/lib --extra-libs='-lpthread -lm' --ld=g++ --bindir=/root/bin --enable-gpl --enable-gnutls --enable-libass --enable-libfdk-aac --enable-vaapi --enable-libsrt --enable-libfreetype --enable-libmp3lame --enable-libopus --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libx265 --enable-nonfree
  libavutil      56. 70.100 / 56. 70.100
  libavcodec     58.134.100 / 58.134.100
  libavformat    58. 76.100 / 58. 76.100
  libavdevice    58. 13.100 / 58. 13.100
  libavfilter     7.110.100 /  7.110.100
  libswscale      5.  9.100 /  5.  9.100
  libswresample   3.  9.100 /  3.  9.100
  libpostproc    55.  9.100 / 55.  9.100
Supported file protocols:
Input:
  async
  cache
  concat
  crypto
  data
  ffrtmphttp
  file
  ftp
  gopher
  gophers
  hls
  http
  httpproxy
  https
  mmsh
  mmst
  pipe
  rtmp
  rtmps
  rtmpt
  rtmpts
  rtp
  srtp
  subfile
  tcp
  tls
  udp
  udplite
  unix
  srt
Output:
  crypto
  ffrtmphttp
  file
  ftp
  gopher
  gophers
  http
  httpproxy
  https
  icecast
  md5
  pipe
  prompeg
  rtmp
  rtmps
  rtmpt
  rtmpts
  rtp
  srtp
  tee
  tcp
  tls
  udp
  udplite
  unix
  srt
```

**Comment by derrod** [comment]
Have you compiled OBS against the self-built ffmpeg and made sure it wasn't linked against an older version?

**Comment by enyone** [comment]
Yes, with instructions listed here https://obsproject.com/wiki/install-instructions#debian-based-build-directions
and by using https://github.com/obsproject/obs-studio/archive/refs/tags/27.0.1.tar.gz instead of git master branch.

```
cmake -DUNIX_STRUCTURE=1 -DFFmpegPath=${HOME}/ffmpeg_build/include/ -DENABLE_PIPEWIRE=OFF -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_BROWSER=OFF -DBUILD_VST=OFF -DCEF_ROOT_DIR="../../cef_binary_4280_linux64" ..
```

Behavior is the same. Does not work (playing stream does not work).

**Comment by enyone** [comment]
Managed to get it working. Problem happened to be my `ffmpeg` built missed shared libraries (./configure --enable-shared ...) and my ffmpeg binaries were actually statically linked ones. For this reason even if ffmpeg binaries seemed to work, OBS was not having correct (wrong ones built without libsrt) shared objects in hand.

Now that OBS is linking to correct shared objects from ffmpeg build with --enable-libsrt set, playing stream using Media Source with srt:// prefix url **does work**.

Note to everyone else: It is not enough to have ffmpeg binaries working as OBS is not using them but instead uses ffmpeg shared objects (libavcodec.so libavdevice.so libavfilter.so libavformat.so libavutil.so libpostproc.so libswresample.so libswscale.so). There is also a risk you fool yourselves (like I did) with a statically linked ffmpeg :)

This bug can be closed.
